pallets/parachain-staking/src/weights.rs

```diff
@@ -261,19 +261,14 @@ impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
 			.saturating_add(T::DbWeight::get().reads(9 as Weight))
 			.saturating_add(T::DbWeight::get().writes(7 as Weight))
 	}
-	// If this takes up too much block space, run again on code
-	// - #743 benchmarks post reward payout optimization were 3x lower per collator,
-	// 15x lower per nominator
 	fn active_on_initialize(x: u32, y: u32) -> Weight {
 		(0 as Weight)
-			// Standard Error: 299_000
-			.saturating_add((208_550_000 as Weight).saturating_mul(x as Weight))
-			// Standard Error: 27_000
-			.saturating_add((15_580_000 as Weight).saturating_mul(y as Weight))
-			.saturating_add(T::DbWeight::get().reads(26 as Weight))
-			.saturating_add(T::DbWeight::get().reads((4 as Weight).saturating_mul(x as Weight)))
-			.saturating_add(T::DbWeight::get().writes(16 as Weight))
-			.saturating_add(T::DbWeight::get().writes((4 as Weight).saturating_mul(x as Weight)))
+			// Standard Error: 1_342_000
+			.saturating_add((46_120_000 as Weight).saturating_mul(x as Weight))
+			// Standard Error: 12_000
+			.saturating_add((1_248_000 as Weight).saturating_mul(y as Weight))
+			.saturating_add(T::DbWeight::get().reads((2 as Weight).saturating_mul(x as Weight)))
+			.saturating_add(T::DbWeight::get().writes((2 as Weight).saturating_mul(x as Weight)))
```

Comment:
This is the result of running the `on_active_initialize` benchmark as-is against these mods. I'm leaving it here to give some idea of how much lighter our round changes are now.

I'm currently working on modifying the benchmarks to properly handle our new cases for `on_initialize`.